File: app/src/main/java/com/noteapp/Controll/MainController.java
package com.noteapp.Controll;

import android.content.Context;
import android.widget.Toast;

import com.noteapp.Model.NoteDatabase;

public class MainController {

    private Context mContext;
    private NoteDatabase mDatabase;
    private NotesAdapter mNotesAdapter;

    public MainController(Context context) {
        this.mContext = context;
        this.mDatabase = new NoteDatabase(mContext);
        this.mNotesAdapter = new NotesAdapter(new NotesAdapter.NoteClickListener() { //TODO: Selbe problem wie beim FAB
            @Override
            public void onNoteClick(int position) {
                Toast.makeText(mContext, mNotesAdapter.getNote(position).getTitle() + " is deleted!", Toast.LENGTH_SHORT).show();
                deleteNote(mNotesAdapter.getNote(position).getId());
                updateNotesAdapterList();
            }
        });
    }

    public NoteDatabase provideDatabase() {
        return this.mDatabase;
    }

    public NotesAdapter provideNotesAdapter() {
        return this.mNotesAdapter;
    }

    public void updateNotesAdapterList() {
        this.mNotesAdapter.setNotes(this.mDatabase.getAllNotes());
    }

    private void deleteNote(int id) {
        this.mDatabase.deleteNote(String.valueOf(id));
    }
}

File: noteappwatch/src/main/java/com/noteappwatch/Controll/MainController.kt
package com.noteappwatch.Controll

import android.content.Context
import android.util.Log
import android.widget.Toast
import com.noteappwatch.Controll.NotesAdapter
import com.noteappwatch.Model.Note
import com.noteappwatch.View.MainActivity
import java.util.*


class MainController(private val mActivity: MainActivity){


    private val mContext: Context
    private val mNotesAdapter: NotesAdapter

    init {
        this.mContext = mActivity.applicationContext
        this.mNotesAdapter = NotesAdapter()
    }

    /* /////////////////////////////////
    Provide unique objects
    ///////////////////////////////// */
    fun provideNotesAdapter(): NotesAdapter {
        return this.mNotesAdapter
    }

    fun provideMainActivity(): MainActivity {
        return this.mActivity
    }

    /* /////////////////////////////////
    Bulk note and database methods
    ///////////////////////////////// */

    fun updateNotesAdapterList() {
        //TODO: Get actual List from phone
        var liste: ArrayList<Note> = ArrayList()
        var i = 0

        liste.add(Note(i++,"title0","text0", Date()))
        liste.add(Note(i++,"titel1","text1", Date()))


        this.mNotesAdapter.setNotes(liste)
    }

    /* /////////////////////////////////
    Communication to the phone
    ///////////////////////////////// */


    /* /////////////////////////////////
    Utility methods
    ///////////////////////////////// */
    fun eventLogInfo(func: String, event: String) {
        Log.println(Log.INFO, func, event)
    }

    fun eventToast(event: String) {
        Toast.makeText(mContext, event, Toast.LENGTH_SHORT).show()
    }
}
